src/servers/lsp.rs

```diff
@@ -195,9 +196,22 @@ impl LspServer {
                 Ok(None)
             }
             "textDocument/didClose" => {
-                let _params: DidCloseTextDocumentParams =
+                let params: DidCloseTextDocumentParams =
                     notification.extract("textDocument/didClose")?;
-                // TODO: remove file?
+
+                let file_id = self
+                    .files
+                    .lock()
+                    .unwrap()
+                    .find_uri(params.text_document.clone())
+                    .unwrap_or_else(|| {
```

Comment:
Possibly just turn this into an `.expect("Could not find file id for document")`. The `unwrap_or_else` with a panic is a bit weird to me.

Reply:
Clippy complains about the `expect` and suggests the `unwrap_or_else` version. I think this is why the `unwrap_or_else` version is previously preferred in the `run_server` method as well.